__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -2434,6 +2435,40 @@ describe('destroyAuthority', () => {
   });
 });
 
+describe('create token with data outputs', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should create a token with data outputs', async () => {
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet, await hWallet.getAddressAtIndex(0), 10);
+    const tx = await createTokenHelper(
+      hWallet,
+      'Token with data outputs',
+      'DOUT',
+      100,
+      {
+        data: ['test1', 'test2']
+      }
+    );
+
+    // Make sure the last 2 outputs are the data outputs
+    const lastOutput = tx.outputs[tx.outputs.length - 1];
+    expect(lastOutput.value).toBe(1);
+    expect(lastOutput.tokenData).toBe(0);
+    const lastOutputScript = parseScriptData(lastOutput.script);
+    expect(lastOutputScript.data).toBe('test2');
+
+    const outputBeforeLast = tx.outputs[tx.outputs.length - 2];
+    expect(outputBeforeLast.value).toBe(1);
+    expect(outputBeforeLast.tokenData).toBe(0);
+    const outputBeforeLastScript = parseScriptData(outputBeforeLast.script);
+    expect(outputBeforeLastScript.data).toBe('test1');
+  });
```

Comment:
I suggest adding another step ( or another test ) using the `CreateTokenTransaction` model `validateNft` helper method on this transaction.

This extra step would ensure that this transaction will not be incorrectly classified as an NFT by our external service.

This assertion would apply for this current solution proposed by this PR and also for the final solution that will be developed, because both would continue relying on the `validateNft` function.

A sample of code that already does this validation can be found on the Wallet Service repository ( [link](https://github.com/HathorNetwork/hathor-wallet-service/blob/ba42e76abbf4064c66feffc8bc97a0deb9d8b9bd/src/utils/nft.utils.ts#L38) )

Reply:
Done https://github.com/HathorNetwork/hathor-wallet-lib/pull/596/commits/010f3ae5d911a6b6263aef378d038e58e0a44663